main/src/ca/uwaterloo/flix/language/phase/unification/Substitution.scala

```diff
@@ -177,6 +189,7 @@ case class Substitution(m: Map[Symbol.KindedTypeVarSym, Type]) {
       // minimize case set formulas if present
       val tpe = x.kind match {
         case Kind.CaseSet(sym) => SetFormula.minimizeType(this.apply(t), sym, sym.universe, SourceLocation.Unknown)
+        case Kind.Eff => TypeMinimization.minimizeType(this.apply(t))
```

Comment:
I would like this small change to be in a separate PR -- to gauge performance cost. Can you make one just for Substitution?